### ISSUE TITLE
App crashes when tracking DerivedState

### ISSUE THREAD

**Issue by lulumeya** [jisungbin/ComposeInvestigator#123]
Reproduce Step : Just launch app with this plugin applied.
composeBomVersion = "2023.10.01"
composeCompiler = "1.5.10"

Stacktrace: 

`java.lang.RuntimeException: Unable to resume activity {com.gangnam.sister.debug/com.kotlin.activity.MainActivity}: java.lang.ClassCastException: androidx.compose.runtime.DerivedSnapshotState cannot be cast to androidx.compose.runtime.snapshots.SnapshotMutableState
                                                                                                    	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5426)
                                                                                                    	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5507)
                                                                                                    	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
                                                                                                    	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:180)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8913)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
                                                                                                    Caused by: java.lang.ClassCastException: androidx.compose.runtime.DerivedSnapshotState cannot be cast to androidx.compose.runtime.snapshots.SnapshotMutableState
                                                                                                    	at land.sungbin.composeinvestigator.runtime.ComposeStateObjectValueGetter.getCurrentValue(StateObjectTracker.kt:234)
                                                                                                    	at land.sungbin.composeinvestigator.runtime.ComposeStateObjectValueGetter.initialize$runtime_release(StateObjectTracker.kt:240)
                                                                                                    	at land.sungbin.composeinvestigator.runtime.StateObjectTrackerKt$registerStateObjectTracking$1$register$2$1.onRemembered(StateObjectTracker.kt:159)
                                                                                                    	at androidx.compose.runtime.CompositionImpl$RememberEventDispatcher.dispatchRememberObservers(Composition.kt:1295)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.applyChangesInLocked(Composition.kt:984)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.applyChanges(Composition.kt:1005)
                                                                                                    	at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(Recomposer.kt:1099)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.composeInitial(Composition.kt:633)
                                                                                                    	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:619)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:123)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:114)
                                                                                                    	at androidx.compose.ui.platform.AndroidComposeView.setOnViewTreeOwnersAvailable(AndroidComposeView.android.kt:1289)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition.setContent(Wrapper.android.kt:114)
                                                                                                    	at androidx.compose.ui.platform.WrappedComposition.onStateChanged(Wrapper.android.kt:164)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.kt:314)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.kt:251)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.kt:287)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.kt:136)
                                                                                                    	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.kt:119)
                                                                                                    	at androidx.fragment.app.FragmentViewLifecycleOwner.handleLifecycleEvent(FragmentViewLifecycleOwner.java:100)
                                                                                                    	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:708)
                                                                                                    	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:3176)
                                                                                                    	at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:3161)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:619)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:275)
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1943)
2024-03-05 13:21:40.287 26853-26853 AndroidRuntime          com.gangnam.sister.debug             E  	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1839)
                                                                                                    	at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1751)
                                                                                                    	at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:323)
                                                                                                    	at androidx.fragment.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:249)
                                                                                                    	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
                                                                                                    	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
                                                                                                    	at androidx.viewpager.widget.ViewPager.setAdapter(ViewPager.java:540)
                                                                                                    	at com.kotlin.activity.MainActivity.initAdapter(MainActivity.kt:601)
                                                                                                    	at com.kotlin.presenter.MainPresenter.onResume(MainPresenter.kt:78)
                                                                                                    	at com.kotlin.activity.MainActivity.onResume(MainActivity.kt:561)
                                                                                                    	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1603)
                                                                                                    	at android.app.Activity.performResume(Activity.java:9119)
                                                                                                    	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5399)
                                                                                                    	... 13 more`

**Comment by jisungbin** [comment]
Thank you for reporting. I will hotfix and release a new version today.

**Comment by jisungbin** [comment]
This issue was caused by trying to track DerivedState, so I added the corresponding code and released it as `1.5.10-0.1.1-SNAPSHOT`. 

Since it is not yet stable released, please add Snapshot Maven as follows:

```gradle
repositories {
  // ...
  maven {
    url 'https://oss.sonatype.org/content/repositories/snapshots/'
  }
}
```

**Comment by jisungbin** [comment]
@lulumeya Please try the new version and let us know if the issue is resolved.

**Comment by lulumeya** [comment]
Did you published the artifact? My build script could not resolve the dependency. (I surely added the snapshot repo)

**Comment by jisungbin** [comment]
> Did you publish the artifact? My build script could not resolve the dependency. (I surely added the snapshot repo)

Did you also add it to pluginManagement's repository? The links below are examples of my application.

- https://github.com/jisungbin/AndroidPlayground/blob/eedfe482d54e58b8f2388423bfd1408cc9f19663/settings.gradle#L7
- https://github.com/jisungbin/AndroidPlayground/blob/eedfe482d54e58b8f2388423bfd1408cc9f19663/settings.gradle#L27
- https://github.com/jisungbin/AndroidPlayground/blob/eedfe482d54e58b8f2388423bfd1408cc9f19663/app/build.gradle#L9

**Comment by jisungbin** [comment]
If all goes well, I plan to release 1.5.10-0.2.0 this week with additional features.

**Comment by lulumeya** [comment]
<img width="1753" alt="image" src="https://github.com/jisungbin/ComposeInvestigator/assets/364983/f2cb8dd1-d3ef-4845-bcd3-674244e26ed1">

I tried new version but there's no luck.
It looks like the plugin dependency causes my dependency graph changes a lot.
Thus it's difficult to address the exact problem.

**Comment by jisungbin** [comment]
@lulumeya Issues that make significant changes to the dependency graph are outside the scope of this issue, so please create a separate issue.

Anyway, to awnser your comment, no dependencies changed between 0.1.0 and 0.1.1-SNAPSHOT, so you need to confirm that the dependency graph changes are indeed caused by applying 0.1.1-SNAPSHOT.

And did applying 0.1.1-SNAPSHOT resolve this issue? If it did, I'll be release 0.1.1 stable.

**Comment by lulumeya** [comment]
The issue not resolved and the dependency graph changes might be cause of this issue. So I mentioned it because the compose API rapidly changes. 
I think, it will be better to release the plugin separately by the versions of Compose regarding this issue.

**Comment by jisungbin** [comment]
This project also depends a bit on the version of the [Compose Runtime](https://developer.android.com/jetpack/androidx/releases/compose-runtime), so we'll take your advice and specify the tested Compose Runtime version in the documentation. Thanks!

And the last tested (0.1.1-SNAPSHOT) Compose Runtime version is 1.6.2 (currently the latest version), and the Compose Runtime version provided in the Compose BOM (2023.10.01) you are using is 1.5.4. I think the different Compose Runtime versions may have an effect.

**Comment by jisungbin** [comment]
Additionally, I tested it by applying the 2023.10.01 BOM with 0.1.1-SNAPSHOT, and this issue appears to have been resolved.

![image](https://github.com/jisungbin/ComposeInvestigator/assets/40740128/293edc17-b91d-4fe3-8fb1-3daf7e55d71f)

If you downgrade to 0.1.0 (same BOM version), the issue will be reproduced.

![image](https://github.com/jisungbin/ComposeInvestigator/assets/40740128/777e5113-a747-4955-b6ad-21ae1dc0b138)

If this issue still occurs, please open a new issue with a new stacktrace. It appears to be resolved now, so I'll release 0.1.1.
